### PR TITLE
Update virions and v1.16 protocol changes

### PR DESCRIPTION
**Pull request by Muqsit**
This PR updates Commando (to v2.1.1) and InvMenu (to v4.0.0) and supplies `causedByRider: false` parameter to `EntityLink` (v1.16).

### CODE REVIEW

File: src/xenialdan/Backpack/BackpackInventory.php [hunk 1]
```diff
@@ -22,9 +23,20 @@ public function getVaultData(): Backpack
 		return $this->vault_data;
 	}
 
-	public function getName(): string
```
Comment (inxomnyaa) [inline src/xenialdan/Backpack/BackpackInventory.php:25]: So no more custom names? rip

File: src/xenialdan/Backpack/BackpackMenuMetadata.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace xenialdan\Backpack;
+
+use muqsit\invmenu\inventory\InvMenuInventory;
+use muqsit\invmenu\metadata\SingleBlockMenuMetadata;
+
+class BackpackMenuMetadata extends SingleBlockMenuMetadata
```
Comment (inxomnyaa) [inline src/xenialdan/Backpack/BackpackMenuMetadata.php:10]: lol wth is this

File: src/xenialdan/Backpack/Loader.php [hunk 3]
```diff
@@ -97,7 +102,7 @@ public function onLoad()
 		Backpack::init();
 
 		Entity::registerEntity(BackpackEntity::class, true, ['backpack']);
-		$this->getServer()->getCommandMap()->register("Backpack", new Commands("backpack", "Manage your backpack"));
+		$this->getServer()->getCommandMap()->register("Backpack", new Commands($this, "backpack", "Manage your backpack"));
```
Comment (inxomnyaa) [inline src/xenialdan/Backpack/Loader.php:105]: eww plugin instance as param, jokes on Cortex. self::$instance is king 🦾

File: src/xenialdan/Backpack/BackpackInventory.php (hunk 1 above)
Comment (Muqsit) [reply src/xenialdan/Backpack/BackpackInventory.php:25]: They work but Inventory::getName() has been unused for a long time, InvMenu::setName still works (https://github.com/Muqsit/Backpack/blob/85575cb0be23e9231f27b936f71e56ffbd676f68/src/xenialdan/Backpack/Backpack.php#L52)

File: src/xenialdan/Backpack/Loader.php [hunk 4]
```diff
@@ -219,6 +224,19 @@ public function onEnable()
 	{
 		@mkdir($this->getDataFolder() . "players");
 		$this->getServer()->getPluginManager()->registerEvents(new EventListener(), $this);
+
+		if (!PacketHooker::isRegistered()) {
```
Comment (inxomnyaa) [inline src/xenialdan/Backpack/Loader.php:228]: Still dont understand why plugins MUST check this.. like for real, nobody gives a f on this exception
register could just return false or sth when already registered

File: src/xenialdan/Backpack/BackpackMenuMetadata.php (hunk 2 above)
Comment (Muqsit) [reply src/xenialdan/Backpack/BackpackMenuMetadata.php:10]: New API method to specify inventory, MenuMetadata holds the graphical stuff that needs to be sent

File: src/xenialdan/Backpack/Loader.php [hunk 5]
```diff
@@ -219,6 +224,19 @@ public function onEnable()
...
+			PacketHooker::register($this);
+		}
+		if (!InvMenuHandler::isRegistered()) {
+			InvMenuHandler::register($this);
+		}
+		if (InvMenuHandler::getMenuType(self::MENU_TYPE_BACKPACK_CHEST) === null) {
+			$copy = InvMenuHandler::getMenuType(InvMenu::TYPE_CHEST);
```
Comment (inxomnyaa) [inline src/xenialdan/Backpack/Loader.php:235]: why the copy?

File: src/xenialdan/Backpack/Loader.php (hunk 3 above)
Comment (Muqsit) [reply src/xenialdan/Backpack/Loader.php:105]: i kind of thought that too, reusing self::$instance would be nice ¯\_(ツ)_/¯

**Comment by inxomnyaa** [comment]
Imagine getting (useful) PRs in 2020. How comes? ^^

File: src/xenialdan/Backpack/Loader.php (hunk 4 above)
Comment (Muqsit) [reply src/xenialdan/Backpack/Loader.php:228]: yeah it's more of an issue when phars are packaged, which is when it's almost always false (exception being if you /reload server)

File: src/xenialdan/Backpack/Loader.php (hunk 3 above)
Comment (inxomnyaa) [reply src/xenialdan/Backpack/Loader.php:105]: @cortexpe plz fix

File: src/xenialdan/Backpack/Loader.php (hunk 5 above)
Comment (Muqsit) [reply src/xenialdan/Backpack/Loader.php:235]: BackpackMenuMetadata holds similar data to the MenuMetadata that InvMenu::TYPE_CHEST is mapped to, i guess $parent makes more sense

**Pull request merged**